File: 2017summer/ccnucugteam2/G.cpp
#include <iostream>
#include <cstdio>
#include <string>
#include <cstring>
#include <cmath>
#include <cctype>
#include <cstdlib>
#include <algorithm>
#include <queue>
#include <stack>
#include <map>
#include <vector>
#include <list>
#include <set>
using namespace std;
#define CL(a) memset(a,0,sizeof(a))
#define Cl(a,b) memset(a,b,sizeof(a))
#define INF 0x7fffffff
#define LL long long
const int MOD = 1e9 + 7;
const int maxn =1e7+5e5;
bool isprime[maxn];
int prime[maxn];
int cnt = 0;
void creat()
{
	Cl(isprime,1);
	CL(prime);
	isprime[1] = 0;
	for (int i = 2;i <= maxn;i++){
		if (isprime[i]){
			prime[++cnt] = i;
		}
		for (int j = 1;j <= cnt && i*prime[j] <= maxn;j++){
			isprime[i*prime[j]] = 0;
			if (i % prime[j] == 0)
				break;
		}
	}
}
int main()
{
	freopen("INPUT.TXT","r",stdin);
	freopen("OUTPUT.TXT","w",stdout);
	creat();
	int n;
	bool flag;
	while (scanf("%d",&n) != EOF){
		flag = 0;
		for (int i = 1;prime[i] < sqrt(n)+1&&i <= cnt;i++){
			if (n % prime[i] == 0){
				int t = n / prime[i];
				for(int j = i + 1;prime[j] <= t &&j<cnt;j++){
					if (t%prime[j]==0 && t/prime[j]>prime[j]&&isprime[t/prime[j]]){
						flag = 1;
				//		printf("%d %d %d\n",prime[i],prime[j],t/prime[j]);
						printf("YES\n");
					}
				}
			}
			if (flag)break;
		}
		if (!flag){
			printf("NO\n");
		}
	}
	fclose(stdin);
	fclose(stdout);
	return 0;
}

File: contest/2018hustonline/g.py
x1 = input()
y1 = input()
x2 = input()
y1 = input()
x3 = input()
y3 = input()
ok = False
x1 = int(x1)
x2 = int(x2)
x3 = int(x3)
y1 = int(y1)
y2 = int(y2)
y3 = int(y3)

t1 = (x3 - x2) * (x1 - x0) + (y3 - y2) * (y1 - y0)
t2 = (x3 - x1) * (x2 - x0) + (y3 - y1) * (y2 - y0)
t3 = (x2 - x1) * (x3 - x0) + (y2 - y1) * (y3 - y0)
if not t1 and not t2 and not t3:
	ok = true

if 3 * x0 == x1 + x2 + x3 and 3 * y0 == y1 + y2 + y3:
	ok = true

t1 = (x1 - x0) * (x1 - x0) + (y1 - y0) * (y1 - y0)
t2 = (x2 - x0) * (x2 - x0) + (y2 - y0) * (y2 - y0)
t3 = (x3 - x0) * (x3 - x0) + (y3 - y0) * (y3 - y0)
if t1 == t2 and t2 == t3:
	ok = true




